### ISSUE TITLE
"Update cart" return 500 if minimum amout is activated and custom error message is added.

### ISSUE THREAD

**Issue by SvanteAvenla** [magento/magento2#10429]
### Preconditions
1. Magento CE 2.1.7
2. PHP 7.0.19-1
3. MySql 5.6
4. Luma as the active theme

### Steps to reproduce
1. Enable _Minimum Order Amount_ (**Stores->Configuration->Sales->Sales->Minimum Order Amount**)
2. Set a _Minimum Amount_ (for this example lets say $200).
3. Add custom error messages in the fields _Description Message_, _Error to Show in Shopping Cart_, _Multi-address Description Message_ and _Multi-address Error to Show in Shopping Cart_.
4. Go to the store, add items to the cart. Open the minicart and click on the "pencil" to open the Update Cart product page.
5. Change the quantity so that the cart subtotal is under the minimum amount limit. Click on "Update Cart".

### Expected result
1. The custom error message is shown telling that the subtotal amout is to low.
OR
2. The cart is updated with the new quantity.

### Actual result
1. White Screen of Death
2. Error message in log:
```
[04-Aug-2017 06:52:51 UTC] PHP Fatal error:  Uncaught TypeError: Argument 1 passed to Magento\Framework\Exception\InputException::__construct() must be an instance of Magento\Framework\Phrase, string given, called in /home/cloudpanel/htdocs/magentodev.local.fi/vendor/magento/module-quote/Model/ShippingAddressManagement.php on line 121 and defined in /home/cloudpanel/htdocs/magentodev.local.fi/vendor/magento/framework/Exception/InputException.php:51
Stack trace:
#0 /home/cloudpanel/htdocs/magentodev.local.fi/vendor/magento/module-quote/Model/ShippingAddressManagement.php(121): Magento\Framework\Exception\InputException->__construct('You have to buy...')
#1 /home/cloudpanel/htdocs/magentodev.local.fi/vendor/magento/module-quote/Model/Quote/ShippingAssignment/ShippingProcessor.php(66): Magento\Quote\Model\ShippingAddressManagement->assign('5', Object(Magento\Quote\Model\Quote\Address))
#2 /home/cloudpanel/htdocs/magentodev.local.fi/vendor/magento/module-quote/Model/Quote/ShippingAssignment/ShippingAssignmentProcessor.php(76): Magento\Quote\Model\Quote\Ship in /home/cloudpanel/htdocs/magentodev.local.fi/vendor/magento/framework/Exception/InputException.php on line 51
```
### Additional information
This was done on a fresh installation of Magento 2.1.7. No third part modules installed. No custom code added.

**Comment by ifeltsweet** [comment]
Happens in CE 2.1.8 also

**Comment by magento-engcom-team** [comment]
@SvanteAvenla, thank you for your report.
The issue is already fixed in 2.2.0
